gateway-ha/src/test/java/io/trino/gateway/CustomImageNameSubstitutor.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.gateway;
+
+import org.testcontainers.utility.DockerImageName;
+import org.testcontainers.utility.ImageNameSubstitutor;
+
+public class CustomImageNameSubstitutor
```

Comment:
The class name is misleading. Please rename to `CustomTrinoImageNameSubstitutor`.

Reply:
From [the doc](https://java.testcontainers.org/features/image_name_substitution/#developing-a-custom-function-for-transforming-image-names-on-the-fly) I suppose we can only have one ImageNameSubstitutor, then we use `TESTCONTAINERS_IMAGE_SUBSTITUTOR=com.mycompany.testcontainers.ExampleImageNameSubstitutor` to tell testcontainer to use this class. Therefore, I think this class should have a generic name as this is the only substitutor that testcontainer will look at.

Reply:
I still disagree with the current name.

Reply:
If the class CustomImageNameSubstitutor is used for other images, `CustomImageNameSubstitutor` seems right but since this is only applied to trino image, I think it's better off with what ebyhr saids. We can change the name to a more general one if there are other images to substitute in future

Reply:
> I still disagree with the current name. Even CustomImageNameSubstitutor doesn't explain whether TESTCONTAINERS_IMAGE_SUBSTITUTOR supports single substitutor or several substitutors either.
> 
> I don't understand your concern to be honest. There's only one class extending ImageNameSubstitutor in this repository. Who will try setting several substitutors?

The concern is we can only have one ImageSubstitutor. Naming it to `CustomTrinoImageNameSubstitutor` is misleading because it might lead to thinking it's only for substituting Trino image. However, this class should be flexible to substitute other images as well. I agree with @Chaho12's suggestion. We can change the name to be generic later as we need.